dds/idl/metaclass_generator.cpp

```diff
@@ -338,6 +338,26 @@ namespace {
     }
     be_global->impl_ << "    }\n";
   }
+
+  void gen_isDcpsKey(IDL_GlobalData::DCPS_Data_Type_Info* info)
+  {
+    if (info && info->key_list_.size()) {
+      for (
+        IDL_GlobalData::DCPS_Key_List::CONST_ITERATOR i(info->key_list_);
+        !i.done(); i.advance()
+      ) {
+        ACE_TString* key;
+        i.next(key);
+        be_global->impl_ <<
+          "    if (!ACE_OS::strcmp(field, \"" << *key << "\")) {\n"
```

Comment:
This is a narrow-character stream so the string in key is going to need conversion in wchar builds.